test/physics/em/MuBremsstrahlung.test.cc

```diff
@@ -0,0 +1,223 @@
+                  gamma.energy.value());
+        EXPECT_LT(0, gamma.energy.value());
+        EXPECT_SOFT_EQ(1.0, celeritas::norm(gamma.direction));
+
+        // Check conservation between primary and secondaries
+        // To be determined: Not sure if momentum is conserved.
+        // this->check_conservation(interaction);
+        this->check_energy_conservation(interaction);
+    }
+
+  protected:
+    celeritas::detail::MuBremsstrahlungInteractorPointers pointers_;
+};
+
+//---------------------------------------------------------------------------//
+// TESTS
+//---------------------------------------------------------------------------//
+
+TEST_F(MuBremsstrahlungInteractorTest, basic) 
+{
+    // Reserve 4 secondaries
+    int num_samples = 4;
+    this->resize_secondaries(num_samples);
+
+    auto material = this->material_track().material_view();
+
+    // Create the interactor
+    MuBremsstrahlungInteractor interact(pointers_,
+                                        this->particle_track(),
+                                        this->direction(),
+                                        this->secondary_allocator(),
+                                        material,
+                                        celeritas::ElementComponentId{0});
+    RandomEngine&              rng_engine = this->rng();
+
+    std::vector<double> energy;
+    std::vector<double> costheta;
+
+    // Produce four samples from the original incident energy
+    for (int i : celeritas::range(num_samples))
+    {
+        Interaction result = interact(rng_engine);
+        SCOPED_TRACE(result);
+        this->sanity_check(result);
+
+        EXPECT_EQ(result.secondaries.data(),
+                  this->secondary_allocator().get().data() + i);
+
+        energy.push_back(result.secondaries[0].energy.value());
+        costheta.push_back(celeritas::dot_product(
+            result.secondaries.front().direction, this->direction()));
+    }
+
+    EXPECT_EQ(num_samples, this->secondary_allocator().get().size());
+
+    // Note: these are "gold" values based on the host RNG.
+    const double expected_energy[]
+        = {1012.99606184083, 1029.80705246907, 1010.52595539471, 1010.77666768483};
+    const double expected_costheta[]
+        = {0.968418002240112, 0.999212413725981, 0.998550042495312, 0.983614606590488};
+
+    EXPECT_VEC_SOFT_EQ(expected_energy, energy);
+    EXPECT_VEC_SOFT_EQ(expected_costheta, costheta);
+
+    // Next sample should fail because we're out of secondary buffer space
+    {
+        Interaction result = interact(rng_engine);
+        EXPECT_EQ(0, result.secondaries.size());
+        EXPECT_EQ(celeritas::Action::failed, result.action);
+    }
+}
+
+
+TEST_F(MuBremsstrahlungInteractorTest, stress_test)
+{
+    const unsigned int num_samples = 1e4;
+    std::vector<double> avg_engine_samples;
+
+    for (auto particle : {pdg::mu_minus(), pdg::mu_plus()})
+    {
+        for (double inc_e : {1.5e4, 5e4, 10e4, 50e4, 100e4})
```

Comment:
Does the value of `avg_engine_samples` becomes too high to be reasonable as it approaches the model limits (1e3 or 1e7 MeV)? It's important to know if we need to worry about having a never ending sampling loop during the interactor's `do {...} while (...);` in edge cases.

Reply:
I tried values that approach the model limits, and it seems that avg_engine_samples are still reasonable.